### ISSUE TITLE
Adaptive Loss

### ISSUE THREAD

**Issue by RomeoPassaro** [kyungmnlee/dmf#1]
Hi, nice work, and thanks for sharing the code!

I have a question regarding the implementation of the adaptive loss.

In the paper, the adaptive loss is written as
$\log\left(\frac{L_{t,r}}{\exp(u_{t,r})}+1\right)+\frac{1}{2}u_{t,r}$

In the implementation, however, I found:

```python
def log_lv_loss(x, y, lv, eps=0.01):
    err = ((x - y) ** 2)
    mse_loss = torch.sum(err, dim=list(range(1, len(x.shape))))
    mean_loss = torch.mean(err, dim=list(range(1, len(x.shape))))
    log_loss = torch.log((1 / lv.exp()) * mean_loss + eps) + lv
    return mse_loss, log_loss
```

This seems to correspond to
$\log\left(\frac{L_{t,r}}{\exp(u_{t,r})}+\epsilon\right)+u_{t,r}$
or equivalently
$\log\left(L_{t,r}+\epsilon\cdot\exp(u_{t,r})\right)$

Unless I am missing something, this appears to differ from the expression in the paper. In particular, it seems that the optimisation dynamics encourage $u_{t,r}$ to decrease without bound, effectively collapsing the adaptive weighting toward a log-loss–like behaviour without an offset.

Am I missing an additional constraint, normalization, or parameterization detail elsewhere in the training code?

Thanks a lot!
